adpcm-decoder/main.c

```diff
@@ -14,14 +15,34 @@ int main(int argc, char* argv[]) {
 
   FILE* in = fopen(argv[1], "rb");
 
-  // Read wave header
+  char riff_id[4];
+  fread(riff_id, 4, 1, in);
+  if (strncmp(riff_id, "RIFF", 4) != 0) {
+    fprintf(stderr, "Invalid RIFF Id: %s\n", riff_id);
+    return 1;
+  }
 
-  // FIXME: Don't skip here, progress here
-  // 12 = chunk type ('fmt ')
-  // 16 = chunk  size (uint32_t chunk_size = 20)
-  fseek(in, 20, SEEK_SET);
+  uint32_t riff_size;
+  fread(&riff_size, 4, 1, in);
+  printf("RIFF size: %d\n", riff_size);
 
-  // Parse 'fmt ' chunk
+  char wave_id[4];
+  fread(wave_id, 4, 1, in);
+  if (strncmp(wave_id, "WAVE", 4) != 0) {
+    fprintf(stderr, "Invalid WAVE Id: %s\n", wave_id);
+    return 1;
+  }
+
+  char fmt_id[4];
+  fread(fmt_id, 4, 1, in);
+  if (strncmp(fmt_id, "fmt ", 4) != 0) {
+    fprintf(stderr, "Invalid fmt Id: %s\n", fmt_id);
```

Comment:
I believe this is a chunk? So it should probably say something like: "Unexpected chunk: '%.4s'. Expected 'fmt '\n" or something.

Make sure you use `%.4s` so the text is cut-off after 4 symbols